repo-settings.c

```diff
@@ -70,8 +71,11 @@ void prepare_repo_settings(struct repository *r)
 		r->settings.fetch_write_commit_graph = value;
 	UPDATE_DEFAULT_BOOL(r->settings.fetch_write_commit_graph, 0);
 
-	if (!repo_config_get_bool(r, "feature.experimental", &value) && value)
+	if (!repo_config_get_bool(r, "feature.experimental", &value) && value) {
 		UPDATE_DEFAULT_BOOL(r->settings.fetch_negotiation_algorithm, FETCH_NEGOTIATION_SKIPPING);
+		if (feature_many_files)
```

Comment:
Do we need to see if this platform supports a FSMonitor backend ??

Reply:
Yes!

Reply:
I forgot to mention that I have a`fsmonitor_ipc__is_supported()` function in `fsmonitor-ipc.c` that is always defined and lets callers avoid `#ifdefs`.